ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml

```diff
@@ -159,6 +159,205 @@ objects:
         mkdir -p /tmp/output
         cd /tmp/output
 
+        function restore-cluster-state() {
+          echo "Placing file /etc/rollback-test with contents A"
+          cat > /tmp/machineconfig.yaml <<'EOF'
+          apiVersion: machineconfiguration.openshift.io/v1
+          kind: MachineConfig
+          metadata:
+            labels:
+              machineconfiguration.openshift.io/role: master
+            name: 99-rollback-test
+          spec:
+            config:
+              ignition:
+                config: {}
+                security:
+                  tls: {}
+                timeouts: {}
+                version: 2.2.0
+              networkd: {}
+              passwd: {}
+              storage:
+                files:
+                - contents:
+                    source: data:,A
+                    verification: {}
+                  filesystem: root
+                  mode: 420
+                  path: /etc/rollback-test
+              systemd: {}
+        EOF
+          oc create -f /tmp/machineconfig.yaml
+
+          function wait_for_machineconfigpool_to_apply() {
+
+            for i in $(seq 0 10); do oc wait machineconfigpool/master --for=condition=Updating --timeout=5m && break; done
+            for i in $(seq 0 10); do oc wait machineconfigpool/master --for=condition=Updated --timeout=5m && break; sleep 30; done
+          }
+
+          wait_for_machineconfigpool_to_apply
+
+          echo "Setting up ssh bastion"
+          mkdir -p ~/.ssh || true
+          cp "${KUBE_SSH_KEY_PATH}" ~/.ssh/id_rsa
+          chmod 0600 ~/.ssh/id_rsa
+          echo "${USER:-default}:x:$(id -u):$(id -g):Default User:$HOME:/sbin/nologin" >> /etc/passwd
+          curl https://raw.githubusercontent.com/eparis/ssh-bastion/master/deploy/deploy.sh | bash
+          BASTION_HOST=$(oc get service -n openshift-ssh-bastion ssh-bastion -o jsonpath='{.status.loadBalancer.ingress[0].hostname}')
+
+          function bastion_ssh() {
+            while true
+            do
+              ssh -o LogLevel=error -o ConnectionAttempts=100 -o ConnectTimeout=5 -o StrictHostKeyChecking=no -o ProxyCommand="ssh -A -o StrictHostKeyChecking=no -o LogLevel=error -o ServerAliveInterval=30 -o ConnectionAttempts=100 -o ConnectTimeout=5 -W %h:%p core@${BASTION_HOST}" $@ && break
+              sleep 10
+            done
+          }
+
+          echo "Make etcd backup on first master"
+          FIRST_MASTER=$(oc get node -l node-role.kubernetes.io/master= -o name | head -n1 | cut -d '/' -f 2)
+          cat > /tmp/etcd_backup.sh <<'EOF'
+            #!/bin/bash
```

Comment:
More here-doc dedenting needed here.

Reply:
I can't use `-EOF` here, as otherwise `$(crictl pods ` would be run locally instead of remote :/
Eventually these scripts would be part of openshift tests (or maybe deployed on every master by the installer)

Reply:
When you can't use `-EOF` (and I'm fine with that), I think you want to dedent content like this line to the level of the EOF.  So here:

```sh
          cat > /tmp/etcd_backup.sh <<EOF
        #!/bin/bash
        ...
        crictl exec ${RUNNING_ETCD_CONTAINER} /bin/sh -c 'source /run/etcd/environment && ETCDCTL_API=3 etcdctl --cert /etc/ssl/etcd/system:etcd-peer:${ETCD_DNS_NAME}.crt --key /etc/ssl/etcd/system:etcd-peer:${ETCD_DNS_NAME}.key --cacert /etc/ssl/etcd/ca.crt snapshot save /var/lib/etcd/snapshot.db'
        EOF
```

Reply:
All these scripts would end up in MCO/installer (see https://github.com/openshift/machine-config-operator/pull/757)